### DISCUSSION TITLE
UI Features

### DISCUSSION THREAD

**Discussion by caleb654** [TNorbury/command_palette#12] [Ideas]
It would be cool to be able to add icons on items and have the option to show instructions for how the user can navigate similar to how Intercom does it 👇

<img width="641" alt="image" src="https://user-images.githubusercontent.com/28836333/173079158-e0302aa1-0c1f-49e3-9a19-a20aee24cca1.png">

**Comment by TNorbury** [comment]
Yeah, that'd be cool. I'll look into adding a `leading` argument for things like icons. For now the best option would be to override `builder` in `CommandPaletteConfig`

**Comment by TNorbury** [comment]
This is out in the latest release `0.5.1`

**Comment by karniv00l** [comment]
Hi, 

I have two more questions/requests:

1. is it possible to add border radius for the top part of the palette? I think it should be consistent for all four corners
2. is there any easy way to add overlay/backdrop with (see example below)? 

<img width="1084" alt="Screenshot 2022-12-11 at 14 16 22" src="https://user-images.githubusercontent.com/3144291/206905892-44c36e60-0dc0-4c1d-a354-470c1dbd2b95.png">

**Reply by TNorbury** [reply]
1. It's currently set with [`CommandPaletteStyle.borderRadius`](https://pub.dev/documentation/command_palette/latest/command_palette/CommandPaletteStyle/borderRadius.html), but verifying just now I realized that it's not working as intended, so I will get a fix out for that.
2. I'd have to look into it. The current grayed out background (set with [`CommandPaletteStyle.commandPaletteBarrierColor`](https://pub.dev/documentation/command_palette/latest/command_palette/CommandPaletteStyle/commandPaletteBarrierColor.html)) is used by the modal's encompassing [`ModalRoute`](https://api.flutter.dev/flutter/widgets/ModalRoute-class.html). I think [`BackdropFilter`](https://api.flutter.dev/flutter/widgets/BackdropFilter-class.html) could maybe do the trick, but I haven't tried it myself. I'll look into it

**Reply by TNorbury** [reply]
Issues:
1. #24
2. #25
